kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -53,13 +54,56 @@ grCollectionsPanel.controller('GrNodeCtrl', ['collections', function(collections
 
 }]);
 
+grCollectionsPanel.controller('GrAddToCollectionCtrl',
+    ['mediaApi', '$scope', function(mediaApi, $scope) {
+
+    const ctrl = this;
+
+    ctrl.saved = false;
+
+    ctrl.addImagesToCollection = imagesJson => {
+        const imageIds = imagesJson.map(imageJson => imageJson.id);
+        const promises = imageIds.map(id => mediaApi.find(id).then(image =>
```

Comment:
To create the resource from the string JSON that we get from the drop data is quite tricky - I will run you through it. Happy to change it later.